pkg/agent/main.go

```diff
@@ -299,6 +299,24 @@ func run() error {
 					if err != nil {
 						logrus.Errorf("failed to check plan: %v", err)
 					}
+
+				case <-ctx.Done():
+					return
+				}
+			}
+		}()
+
+		go func() {
+			for {
+				select {
+				case <-time.After(time.Minute):
```

Comment:
since you are adding the reconcile to the beginning, i think we should increase the retrigger to three minutes